test/unit/lib/plugins/aws/package/compile/events/msk/index.test.js

```diff
@@ -9,11 +9,434 @@ chai.use(require('chai-as-promised'));
+            },
+          };
+        },
+      },
+
+      {
+        name: 'saslScram512Auth',
+        event: {
+          topic,
+          bootstrapServers: ['abc.xyz:9092'],
+          accessConfigurations: [
+            {
+              saslScram512Auth:
+                'arn:aws:secretsmanager:us-east-1:01234567890:secret:SaslScram512SecretName',
+            },
+          ],
+        },
+        resource: (awsNaming) => {
+          return {
+            SelfManagedEventSource: {
+              Endpoints: {
+                KafkaBootstrapServers: ['abc.xyz:9092'],
+              },
+            },
+            SourceAccessConfigurations: [
+              {
+                Type: 'SASL_SCRAM_512_AUTH',
+                URI: 'arn:aws:secretsmanager:us-east-1:01234567890:secret:SaslScram512SecretName',
+              },
+            ],
+            StartingPosition: 'TRIM_HORIZON',
+            Topics: [topic],
+            FunctionName: {
+              'Fn::GetAtt': [awsNaming.getLambdaLogicalId('foo'), 'Arn'],
+            },
+          };
+        },
+      },
+
+      {
+        name: 'unknownType',
+        event: {
+          topic,
+          bootstrapServers: ['abc.xyz:9092'],
+          accessConfigurations: [
+            {
+              UNKNOWN_TYPE: 'arn:aws:secretsmanager:us-east-1:01234567890:secret:UnknownSecretName',
+            },
+          ],
+        },
+        resource: (awsNaming) => {
+          return {
+            SelfManagedEventSource: {
+              Endpoints: {
+                KafkaBootstrapServers: ['abc.xyz:9092'],
+              },
+            },
+            SourceAccessConfigurations: [
+              {
+                Type: 'UNKNOWN_TYPE',
+                URI: 'arn:aws:secretsmanager:us-east-1:01234567890:secret:UnknownSecretName',
+              },
+            ],
+            StartingPosition: 'TRIM_HORIZON',
+            Topics: [topic],
+            FunctionName: {
+              'Fn::GetAtt': [awsNaming.getLambdaLogicalId('foo'), 'Arn'],
+            },
+          };
+        },
+      },
+
+      {
+        name: 'refValue',
+        event: {
+          topic,
+          bootstrapServers: ['abc.xyz:9092'],
+          accessConfigurations: [{ vpcSubnet: { Ref: 'VpcSubnet' } }],
+        },
+        resource: (awsNaming) => {
+          return {
+            SelfManagedEventSource: {
+              Endpoints: {
+                KafkaBootstrapServers: ['abc.xyz:9092'],
+              },
+            },
+            SourceAccessConfigurations: [
+              {
+                Type: 'VPC_SUBNET',
+                URI: { 'Fn::Sub': ['subnet:${value}', { value: { Ref: 'VpcSubnet' } }] },
+              },
+            ],
+            StartingPosition: 'TRIM_HORIZON',
+            Topics: [topic],
+            FunctionName: {
+              'Fn::GetAtt': [awsNaming.getLambdaLogicalId('foo'), 'Arn'],
+            },
+          };
+        },
+      },
+    ].forEach((eventConfig) => {
```

Comment:
This might be my personal prefrence, but did you consider e.g. abstracting away `it` block with test to a function and instead of iterating here, do something like

```
const runCompileEventSourceMappingTest = (eventConfig) => {...}

it('for event xxx', async () => {
const eventConfig = {...}
await runCompileEventSourceMapppingTest(eventConfig)
})
```

The result is similar, but each separate test is a bit easier to grasp (as mentioned, it might be my personal preference and it might not be "easier to grasp" for everyone).

Reply:
I hadn't thought about that, but I do like it.  At least for what I have here, since the payloads are a bit unwieldy.